File: backend/app/schemas/trade.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class TradeResponse(BaseModel):
    id: uuid.UUID
    market_id: str
    market_question: str
    market_category: str | None
    action: str
    side: str
    shares: Decimal
    price: Decimal
    total_amount: Decimal
    confidence_score: Decimal | None
    edge: Decimal | None
    sources_count: int | None
    reasoning: str | None
    enforcement_result: str
    armoriq_plan_hash: str | None
    executed_at: datetime

    model_config = {"from_attributes": True}


class ApprovalResponse(BaseModel):
    id: uuid.UUID
    market_id: str
    market_question: str
    action: str
    side: str
    shares: Decimal
    price: Decimal
    total_amount: Decimal
    category: str | None
    confidence_score: Decimal | None
    reasoning: str | None
    sources: dict | None
    threshold_breached: str | None
    status: str
    expires_at: datetime
    created_at: datetime

    model_config = {"from_attributes": True}


class ApprovalAction(BaseModel):
    note: str | None = None

File: frontend/src/types/trade.ts
export type TradeStatus =
  | 'pending'
  | 'approved'
  | 'held'
  | 'denied'
  | 'executed'
  | 'auto_approved'
  | 'blocked'
  | 'cancelled';

export type TradeSide = 'buy' | 'sell';

export interface Trade {
  id: string;
  market_id: string;
  market_question: string;
  market_category: string | null;
  action: string;
  side: string;
  shares: number;
  price: number;
  total_amount: number;
  confidence_score: number | null;
  edge: number | null;
  sources_count: number | null;
  reasoning: string | null;
  enforcement_result: string;
  armoriq_plan_hash: string | null;
  executed_at: string;
}

export interface ApprovalRequest {
  id: string;
  market_id: string;
  market_question: string;
  action: string;
  side: string;
  shares: number;
  price: number;
  total_amount: number;
  category: string | null;
  confidence_score: number | null;
  reasoning: string | null;
  sources: unknown[] | Record<string, unknown> | null;
  threshold_breached: string | null;
  status: string;
  expires_at: string;
  created_at: string;
}
